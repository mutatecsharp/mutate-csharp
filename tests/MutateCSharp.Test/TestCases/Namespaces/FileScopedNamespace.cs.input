namespace FileScopedNamespace;

public class Program
{
  static void Main()
  {
  }
}