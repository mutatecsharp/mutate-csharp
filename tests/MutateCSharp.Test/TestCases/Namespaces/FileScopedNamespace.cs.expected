namespace MutateCSharp
{
  public static class Schemata
  {
    private static bool _initialised;
    private static int _specifiedMutantId;

    private static void Initialise()
    {
      if (_initialised) return;
      var activatedMutant = Environment.GetEnvironmentVariable("MUTATE_CSHARP_ACTIVATED_MUTANT");
      if (!string.IsNullOrEmpty(activatedMutant)) _specifiedMutantId = int.Parse(activatedMutant);
      _initialised = true;
    }
    
    private static bool ActivatedInRange(int lowerBound, int upperBound)
    {
      Initialise();
      return lowerBound <= _specifiedMutantId && _specifiedMutantId <= upperBound;
    }

    private static bool Activated(int mutantId) => mutantId == _specifiedMutantId;
  }
}

namespace FileScopedNamespace
{
  public class Program
  {
    static void Main()
    {
    }
  }
}
